### PR TITLE
Feature/gen 74 company list integration

### PR DESCRIPTION
**Pull request by SaurabhChandel007**
company

### CODE REVIEW

File: admin-ui/app/company/[id]/page.tsx [hunk 1]
```diff
@@ -1,36 +1,29 @@
...
-}) {
-  const [company, setCompany] = useState<ICompany>();
-  const { data, error, isError, isLoading } = useQuery(
-    [`company-${params?.id}`],
-    async () => await fetchData(params?.id)
-  );
+const page = () => {
+  const [company, setCompany] = useState([]);
+  const fetchData = () => {
+    fetch(`${process.env.NEXT_PUBLIC_API_URL}/companies/${_id}`)
```
Comment (vivekbarsagadey) [inline admin-ui/app/company/[id]/page.tsx:10]: wrong way to load data

File: admin-ui/app/company/company.model.ts [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
-    _id:string;
-    name: string;
-    mobile: string;
-    email:string;
-    address:string
-    country:string
-      state:string
-      city:string
-      pincode:string
+interface ICompany {
```
Comment (vivekbarsagadey) [inline admin-ui/app/company/company.model.ts:1]: can you move to type

File: admin-ui/app/company/create/company.create.tsx [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+  const [ownerLastName, setOwnerLastName] = useState("");
+  const [companyName, setCompanyName] = useState("");
+  const [companyEmail, setCompanyEmail] = useState("");
+  const [companyPhone, setCompanyPhone] = useState("");
+  const [companyAddress, setCompanyAddress] = useState("");
+  const [companyWebsite, setCompanyWebsite] = useState("");
+  const [companyFoundationYear, setCompanyFoundationYear] = useState("");
+  const router = useRouter();
+
+  const updateMyCompanyData = async () => {
```
Comment (vivekbarsagadey) [inline admin-ui/app/company/create/company.create.tsx:18]: why we are not using service

File: admin-ui/app/project/create/index.tsx [hunk 4]
```diff
@@ -3,10 +3,10 @@ import { Button, Grid, Paper, TextField, Typography } from "@mui/material";
 import Checkbox from "@mui/material/Checkbox";
 import Link from "next/link";
 import { useRouter } from "next/navigation";
-import { useState } from "react";
+import { useState, useEffect } from "react";
 import { applicationType } from "../../../component/common/data/project/application.type";
```
Comment (vivekbarsagadey) [inline admin-ui/app/project/create/index.tsx:7]: why do you need this data

File: admin-ui/app/project/create/index.tsx [hunk 5]
```diff
@@ -17,14 +17,36 @@ const ProjectCreate = () => {
...
+      `${process.env.NEXT_PUBLIC_API_URL}/companies`
+    );
+    if (!response.ok) {
+      throw new Error("Data coud not be fetched!");
+    } else {
+      return response.json();
+    }
+  };
+  useEffect(() => {
+    fetchData()
```
Comment (vivekbarsagadey) [inline admin-ui/app/project/create/index.tsx:32]: same data fatching

File: admin-ui/app/project/create/index.tsx [hunk 6]
```diff
@@ -41,7 +63,6 @@ const ProjectCreate = () => {
     }).then((res) => res.json());
```
Comment (vivekbarsagadey) [inline admin-ui/app/project/create/index.tsx:63]: wrong code

File: admin-ui/app/project/list/index.tsx [hunk 7]
```diff
@@ -26,7 +26,7 @@ const ProjectListComponent = ({
 
   return (
     <>
-      <Box mt={2}>
+      <Box mt={2} width={"98.2%"}>
```
Comment (vivekbarsagadey) [inline admin-ui/app/project/list/index.tsx:29]: no need to have %

File: admin-ui/app/project/list/index.tsx [hunk 8]
```diff
@@ -42,7 +42,7 @@ const ProjectListComponent = ({
               <Typography variant="subtitle2">Project Name</Typography>
             </Grid>
             <Grid item xs={2.7} sm={2.7}>
```
Comment (vivekbarsagadey) [inline admin-ui/app/project/list/index.tsx:44]: what is 2.7

File: admin-ui/app/project/list/index.tsx [hunk 9]
```diff
@@ -42,7 +42,7 @@ const ProjectListComponent = ({
...
-              <Typography variant="subtitle2">Customer Name</Typography>
+              <Typography variant="subtitle2">Company Name</Typography>
             </Grid>
             <Grid item xs={2.8} sm={3.4}>
```
Comment (vivekbarsagadey) [inline admin-ui/app/project/list/index.tsx:47]: same 2.8 or 3.4

File: admin-ui/app/utility/company/csv.generator.tsx [hunk 10]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
+import Typography from "@mui/material/Typography";
+import { CSVLink } from "react-csv";
+
+const CompanyCsvGenerator = ({ copyCompanyData }:any) => {
+  return (
+    <Typography variant="subtitle1">
+      <CSVLink
+        data={copyCompanyData}
+        filename={`CompanyData ${new Date().toISOString().slice(0, 10)}`}
```
Comment (vivekbarsagadey) [inline admin-ui/app/utility/company/csv.generator.tsx:10]: format, file name util, it must have company-screenname-date.ext

File: admin-ui/app/utility/company/excell.generator.tsx [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+  "createdAt",
+  "updatedAt",
+  " Project Name",
+  "Customer Name",
+  "Application",
+];
+const CompanyExcellGenerator = ({ copyCompanyData }) => {
+  function handleDownloadExcel() {
+    downloadExcel({
+      fileName: `Project ${new Date().toISOString().slice(0, 10)}`,
```
Comment (vivekbarsagadey) [inline admin-ui/app/utility/company/excell.generator.tsx:16]: same file format issue

File: admin-ui/app/project/create/index.tsx (hunk 4 above)
Comment (SaurabhChandel007) [reply admin-ui/app/project/create/index.tsx:7]: We need this data for different types of Application based on business like 
B2C
B2B
etc..

File: admin-ui/app/company/[id]/page.tsx [hunk 12]
```diff
@@ -1,36 +1,27 @@
...
+  const [company, setCompany] = useState([]);
 
-  if (isLoading) {
-    return <>Please wait .....</>;
-  }
-  if (isError) {
-    return <>Please wait there is some error</>;
-  }
+  const fetchData = async () => {
+    const users = await fetch(
```
Comment (vivekbarsagadey) [inline admin-ui/app/company/[id]/page.tsx:11]: plz check, i think we have to use query, I think this 

 concurrentFeatures: true,

https://learnjsx.com/category/4/posts/nextjs-suspense

File: admin-ui/app/company/[id]/page.tsx [hunk 13]
```diff
@@ -1,36 +1,27 @@
...
+  };
+  useEffect(() => {
+    fetchData();
+  }, []);
 
-  return <CompanyComponent company={data} />;
-}
+  return (
+    <>
+      <CompanyEditComponent company={company} _id={_id} />
```
Comment (vivekbarsagadey) [inline admin-ui/app/company/[id]/page.tsx:23]: dont us _id ... it will be id

File: admin-ui/app/company/filters/index.tsx [hunk 14]
```diff
@@ -1,88 +1,114 @@
...
+import ICompany from "../company.model";
 
-interface FilterComponentProps extends ICompanyComponentProps {
+type CompanyComponentProps = {
   handleClose: () => void;
-  anchorEl: boolean;
+  companyData: Array<ICompany>;
   open: boolean;
-}
+  anchorEl: boolean;
```
Comment (vivekbarsagadey) [inline admin-ui/app/company/filters/index.tsx:14]: this is not required anchorEl , or use proper name

File: admin-ui/app/company/filters/index.tsx [hunk 15]
```diff
@@ -1,88 +1,114 @@
...
   const doFilter = () => {
-    if (filterDataName !== "" && filterDataEmail !== "") {
-      const newCompany = items?.filter(
-        (c) =>
-          c.name.toLowerCase().includes(filterDataName.toLowerCase()) ||
-          c.email.toLowerCase().includes(filterDataEmail.toLowerCase())
-      );
-      itemsCallBackHandler(newCompany);
-    }
+    const companyFilterData = companyData?.filter(
```
Comment (vivekbarsagadey) [inline admin-ui/app/company/filters/index.tsx:30]: filter code need to changes

File: admin-ui/app/company/filters/index.tsx [hunk 16]
```diff
@@ -1,88 +1,114 @@
...
+        </div>
 
-        <Grid container mb={1}>
+        <div>
+          <Stack sx={{ width: 300, padding: "1rem" }}>
+            <Autocomplete
+              size="small"
+              freeSolo
+              disableClearable
+              options={Array.from(new Set(companyData.map((id) => id.email)))}
```
Comment (vivekbarsagadey) [inline admin-ui/app/company/filters/index.tsx:78]: plz check code

File: admin-ui/app/company/filters/index.tsx [hunk 17]
```diff
@@ -1,88 +1,114 @@
...
+                />
+              )}
+              onChange={(event, value) => setFilterDataName(value)}
+            />
+          </Stack>
+        </div>
 
-        <Grid container mb={1}>
+        <div>
+          <Stack sx={{ width: 300, padding: "1rem" }}>
```
Comment (vivekbarsagadey) [inline admin-ui/app/company/filters/index.tsx:73]: change this

**Pull request merged**